#! /usr/bin/env python
PACKAGE = "vesselness_image_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Preprocesing Kernel information:
gen.add("side_h", int_t, 0, "kernel width", 3, 1, 10)
gen.add("variance_h", double_t, 0, "kernel variance", 0.1, 0, 10)

# Processing parameters:
gen.add("beta", double_t, 0, "ratio sensitivity", 0.1, 0, 1)
gen.add("c", double_t, 0, "noise suppression", 0.1, 0, 1)

# post processing information:
gen.add("side_p", int_t, 0, "kernel width", 3, 1, 10)
gen.add("variance_p", double_t, 0, "kernel variance", 0.1,  0, 10)


exit(gen.generate(PACKAGE, "vesselness_params_", "vesselness_params_"))
